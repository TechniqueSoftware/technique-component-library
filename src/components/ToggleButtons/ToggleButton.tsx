import * as React from 'react';
import { default as MuiToggleButton, ToggleButtonProps } from '@material-ui/lab/ToggleButton';

import { Theme } from '@material-ui/core';
import { makeStyles, createStyles } from '@material-ui/styles';

export {
  ToggleButtonProps,
};

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      fontWeight: 'bold',
      textAlign: 'center',
      minWidth: 21,
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1),
    },
  }),
);

function ToggleButton(props: ToggleButtonProps) {
  const {
    children,
  } = props;
  const classes = useStyles({});

  const partialProps = { ...props };
  delete partialProps.children;

  return (
    <MuiToggleButton {...partialProps}>
      {typeof children === 'string'
        ? <div className={classes.content}>{children}</div>
        : children
      }
    </MuiToggleButton>
  );
}

export default ToggleButton;
