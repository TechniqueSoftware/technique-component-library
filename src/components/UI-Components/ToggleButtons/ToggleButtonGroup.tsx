//@ts-nocheck
import * as React from 'react';
import { default as MuiToggleButtonGroup, ToggleButtonGroupProps } from '@material-ui/lab/ToggleButtonGroup';

const ToggleButtonGroup = (props: ToggleButtonGroupProps): JSX.Element => {
  const applyProps: ToggleButtonGroupProps = {
    ...props,
    size: 'small',
  };
  return (
    <MuiToggleButtonGroup {...applyProps}/>
  );
};

export {
  ToggleButtonGroupProps
};

export default ToggleButtonGroup;
