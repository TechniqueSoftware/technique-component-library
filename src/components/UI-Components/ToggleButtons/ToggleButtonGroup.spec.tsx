import { shallow } from 'enzyme';
import * as React from 'react';

import ToggleButton from './ToggleButton';
import ToggleButtonGroup, { ToggleButtonGroupProps } from './ToggleButtonGroup';
import FormatAlignLeftIcon from '@material-ui/icons/FormatAlignLeft';

describe('ToggleButtonGroup', () => {
  test('should render with base props', () => {
    const baseProps: ToggleButtonGroupProps = {
      value: 'left',
      exclusive: true,
      children: (
        <ToggleButton value="left" aria-label="left aligned">
          <FormatAlignLeftIcon/>
        </ToggleButton>
      ),
    };
    const shallowWrapper = shallow(
      <ToggleButtonGroup {...baseProps}/>
    );
    expect(shallowWrapper.debug()).toMatchSnapshot();
  });
});
