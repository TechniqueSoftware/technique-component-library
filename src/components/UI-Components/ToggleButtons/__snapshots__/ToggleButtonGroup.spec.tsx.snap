// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButtonGroup should render with base props 1`] = `
"<WithStyles(ForwardRef(ToggleButton)) value=\\"left\\" exclusive={true} size=\\"small\\">
  <ToggleButton value=\\"left\\" aria-label=\\"left aligned\\">
    <Memo(ForwardRef(FormatAlignLeftIcon)) />
  </ToggleButton>
</WithStyles(ForwardRef(ToggleButton))>"
`;
